sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/BasicWriteStatsTracker.scala

```diff
@@ -192,6 +192,10 @@ class BasicWriteJobStatsTracker(
     new BasicWriteTaskStatsTracker(serializableHadoopConf.value)
   }
 
+  override def processCommitDuration(duration: Long): Unit = {
+    metrics(BasicWriteJobStatsTracker.DURATION_FILE_COMMIT).set(duration)
```

Comment:
Does it really work? The broadcast exchange does not only set the metrics, but also call `postDriverMetricUpdates`.